internal/registration/registration.go

```diff
@@ -3,29 +3,63 @@ package registration
 import (
 	"context"
 	"encoding/json"
+	"git.sr.ht/~spc/go-log"
 	"github.com/google/uuid"
+	"github.com/jakub-dzon/k4e-device-worker/internal/configuration"
 	hardware2 "github.com/jakub-dzon/k4e-device-worker/internal/hardware"
 	os2 "github.com/jakub-dzon/k4e-device-worker/internal/os"
 	"github.com/jakub-dzon/k4e-operator/models"
 	pb "github.com/redhatinsights/yggdrasil/protocol"
 	"time"
 )
 
+const (
+	retryAfter = 10
+)
+
 type Registration struct {
 	hardware         *hardware2.Hardware
 	os               *os2.OS
 	dispatcherClient pb.DispatcherClient
+	config           *configuration.Manager
+	RetryAfter       int64
 }
 
-func NewRegistration(hardware *hardware2.Hardware, os *os2.OS, dispatcherClient pb.DispatcherClient) *Registration {
+func NewRegistration(hardware *hardware2.Hardware, os *os2.OS, dispatcherClient pb.DispatcherClient, config *configuration.Manager) *Registration {
 	return &Registration{
 		hardware:         hardware,
 		os:               os,
 		dispatcherClient: dispatcherClient,
+		config:           config,
+		RetryAfter:       retryAfter,
+	}
+}
+
+func (r *Registration) RegisterDevice() {
```

Comment:
How the agent know that there is no need to talk to the operator if we are not registered? ATM even if we are not registered there is a lot of 404 is the logs for the agent attempts.

Reply:
It doesn't. The gateway - yggdrasil starts polling its communication channels right after start - it can't be controlled by the  workers. I will be making sure that the HTTP errors are not logged on the lowest possible level (error will also be encountered when the device is "away").

Reply:
OK, let's explore with fleet mgmt how we could do it. In our context it doesn't make sense to poll if we are not registered.